src/main/java/jcifs/netbios/NameServiceClientImpl.java

```diff
@@ -0,0 +1,1072 @@
+                    response.addressArray[ i ].hostName.srcHashCode = srcHashCode;
+                }
+                return response.addressArray;
+            }
+        }
+        throw new UnknownHostException(addr.getHostName());
+    }
+
+
+    @Override
+    public NbtAddress getNbtByName ( String host ) throws UnknownHostException {
+        return getNbtByName(host, 0x00, null);
+    }
+
+
+    @Override
+    public NbtAddress getNbtByName ( String host, int type, String scope ) throws UnknownHostException {
+        return getNbtByName(host, type, scope, null);
+    }
+
+
+    @Override
+    public NbtAddress getNbtByName ( String host, int type, String scope, InetAddress svr ) throws UnknownHostException {
+
+        if ( host == null || host.length() == 0 ) {
+            return getLocalHost();
+        }
+
+        Name name = new Name(this.transportContext.getConfig(), host, type, scope);
+        if ( !Character.isDigit(host.charAt(0)) ) {
+            return doNameQuery(name, svr);
+        }
+
+        int IP = 0x00;
+        int hitDots = 0;
+        char[] data = host.toCharArray();
+
+        for ( int i = 0; i < data.length; i++ ) {
+            char c = data[ i ];
+            if ( c < 48 || c > 57 ) {
+                return doNameQuery(name, svr);
+            }
+            int b = 0x00;
+            while ( c != '.' ) {
+                if ( c < 48 || c > 57 ) {
+                    return doNameQuery(name, svr);
+                }
+                b = b * 10 + c - '0';
+
+                if ( ++i >= data.length )
+                    break;
+
+                c = data[ i ];
+            }
+            if ( b > 0xFF ) {
+                return doNameQuery(name, svr);
+            }
+            IP = ( IP << 8 ) + b;
+            hitDots++;
+        }
+        if ( hitDots != 4 || host.endsWith(".") ) {
+            return doNameQuery(name, svr);
+        }
+        return new NbtAddress(getUnknownName(), IP, false, NbtAddress.B_NODE);
+    }
+
+
+    @Override
+    public NbtAddress[] getNbtAllByName ( String host, int type, String scope, InetAddress svr ) throws UnknownHostException {
+        return getAllByName(new Name(this.transportContext.getConfig(), host, type, scope), svr);
+    }
+
+
+    @Override
+    public NbtAddress[] getNbtAllByAddress ( String host ) throws UnknownHostException {
+        return getNbtAllByAddress(getNbtByName(host, 0x00, null));
+    }
+
+
+    @Override
+    public NbtAddress[] getNbtAllByAddress ( String host, int type, String scope ) throws UnknownHostException {
+        return getNbtAllByAddress(getNbtByName(host, type, scope));
+    }
+
+
+    @Override
+    public NbtAddress[] getNbtAllByAddress ( NetbiosAddress addr ) throws UnknownHostException {
+        try {
+            NbtAddress[] addrs = getNodeStatus(addr);
+            cacheAddressArray(addrs);
+            return addrs;
+        }
+        catch ( UnknownHostException uhe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)